activiti-api-task-model/src/main/java/org/activiti/api/task/model/builders/CompleteTaskPayloadBuilder.java

```diff
@@ -30,8 +36,19 @@ public CompleteTaskPayloadBuilder withVariable(String name,
         return this;
     }
 
+    public CompleteTaskPayloadBuilder withTaskVariable(String name,
+                                                       Object value) {
+        if (this.taskVariables == null) {
+            this.taskVariables = new HashMap<>();
+        }
+        this.taskVariables.put(name,
+                           value);
+        return this;
+    }
+    
     public CompleteTaskPayload build() {
         return new CompleteTaskPayload(taskId,
-                                       variables);
+                                       variables,
+                                       taskVariables);
```

Comment:
@igdianov this is extremely confusing to me.. I do understand why.. but as a new user I would be completely lost of why I have two set of variables.. this is not intuitive at all. When things like this pop up are usually bad API design.. I will let @erdemedeiros comment on it.. 
In my head, we can simplify by:
1) Save will store task local variables
2) complete will send only variables that needs to be propagated to the process instance scope (in my head this should check if the variables exist in the process instance, if not discard)

The assumption here is that in the UI, you can save as many times that you want, and when complete you will send only the variables that you want to propagate to the process scope level, so in a way, you need to pick variables from task scope and then consciously choose which ones will be propagated.

Reply:
@salaboy @igdianov, it's not very clear to me which are the use cases to set task variables while completing a task (maybe for history/audit purpose). We have different possibilities based on which use cases to cover:
- Case 1: allow setting both task and process variables while completing:
In this case we need two variable collections as in this PR. What is not really a problem in my opinion, but this case, it's probably better to have clearly `processVariables` and `taskVariables` instead of `variables` and `taskVariables` (breaking change).
- Case 2 (what activiti 6 api does): allow to chose the scope for all the variables, they are all going to be either task variables or all process variables. In this case we just need to add an extra flag `useLocalScopeForVariables`
- Case 3: always use process as variables scope.

Reply:
@erdemedeiros for case 1: I have the problem that we are exposing too much to the user. 
for Case 2: this is not the old API anymore this is trying to solve some of the common problem that the old API has, so I don't want to expose the internals
We can hide the internals behind the APIs and then decide what is the default logic to expose.